### ISSUE TITLE
Improve `mz_materialization_frontiers` to include sources and sinks

### ISSUE THREAD

**Issue by frankmcsherry** [MaterializeInc/materialize#10089]
### Feature request

The introspection collection `mz_materialization_frontiers` contains frontier information (open timestamps) for arrangements. This information allows users to see what the current state of the system reflects, and to look for skew between frontiers, indicating where the system may be holding up work (contributing to end-to-end latency).

We want to improve this to include the boundaries of `dataflow` as well, specifically sources and sinks, so that we can better diagnose end-to-end latency defects. Ideally, one would be able to interrogate the system and observe the frontiers at the sources, those at resulting sinks, and the intervening views, and either confirm or disconfirm that the compute layer is keeping the answers up to date (either accepting responsibility, or pushing back to the source or sink implementation and their integrations, e.g. perhaps Kafka being slow idk).

**Comment by benesch** [comment]
Kicking this one out of **M-platform-blocker**. We've marked the introspection sources as internal-only for EA (#14929).

**Comment by dseisun-materialize** [comment]
Given the rename [here](https://github.com/MaterializeInc/materialize/pull/14934) I'm gathering this can be closed? @teskje did I imagine that you picked up some work around this recently?

**Comment by teskje** [comment]
The rename didn't change anything and the per-replica introspection sources still only expose compute frontiers (indexes, MVs, subscribes). I think this will only change once we have full cluster unification, which as I understand, is still way out.

But we do have `mz_frontiers` and mz_global_frontiers` now, which contain frontiers for all storage and compute objects. So these frontiers are available, just not in the source mentioned in the description of this issue. Seems fine to close it.

**Comment by dseisun-materialize** [comment]
makes sense!

**Comment by benesch** [comment]
> I think this will only change once we have full cluster unification, which as I understand, is still way out.

That's right. I think we'll need to revisit the naming of this relation once we unify the timely clusters, as `mz_compute_frontiers` will suddenly contain frontiers of both storage and compute objects. I added a note about this to #17416.
